javaslang/src/main/java/javaslang/collection/HashMultimap.java

```diff
@@ -25,10 +24,22 @@
  * @author Ruslan Sennov
  * @since 2.1.0
  */
-public final class HashMultimap<K, V> extends AbstractMultimap<K, V, HashMultimap<K, V>> implements Serializable {
+public final class HashMultimap<K, V> implements Multimap<K, V>, Serializable {
 
     private static final long serialVersionUID = 1L;
 
+    private final Map<K, Traversable<V>> back;
```

Comment:
I find `back` to be a confusing name, is it an abbreviation of `backing` or is it meant as the opposite of `front` or does it mean that something is stored in reverse order?
In my opinion `delegate` might be more appropriate in this case.